website/lib/changelog/generate-component-changelog-entries.mjs

```diff
@@ -99,6 +100,75 @@ const updateComponentVersionHistory = (componentChangelogEntries, version) => {
   });
 };
 
+const updateComponentFrontMatter = (componentChangelogEntries, version) => {
+  Object.keys(componentChangelogEntries).forEach((componentName) => {
+    const indexPath = `${allComponentsPath[componentName]}/index.md`;
+
+    if (fs.existsSync(indexPath)) {
+      // Fetch the index markdown file
+      const indexContent = fs.readFileSync(indexPath, 'utf8');
+
+      // Parse the index file content
+      const parsedFrontMatter = matter(indexContent);
+
+      // Update the front matter
+      if (!parsedFrontMatter.data.status) {
+        parsedFrontMatter.data.status = {};
+      }
+      if (
+        parsedFrontMatter.data.status.added !== version &&
+        parsedFrontMatter.data.status.updated !== version
+      ) {
+        parsedFrontMatter.data.status.updated = version;
+      }
+
+      // Stringify the updated content
+      const updatedContent = matter.stringify(
+        parsedFrontMatter.content,
+        parsedFrontMatter.data
+      );
+
+      // Write the updated content back to the index markdown file
+      fs.writeFileSync(indexPath, updatedContent);
+    }
+  });
+};
+
+const cleanComponentFrontMatter = (components) => {
```

Comment:
Suggestion: I think version needs to be added to the args here? Otherwise I don't see where it is defined.

Reply:
right, yes, I believe I'm missing the second argument here – will update!

Reply:
fixed in [15e50a9](https://github.com/hashicorp/design-system/pull/2656/commits/15e50a923b3135793dcd7fe047e9df93aa43b500) thanks for spotting this!